tests/integration/security/testdata/authz/v1beta1-path.yaml.tmpl

```diff
@@ -14,6 +14,6 @@ spec:
   rules:
   - to:
     - operation:
-        paths: ["/public*"]
+        paths: ["/public*", 'regex:\/abc\/[0-9]?']
```

Comment:
let's use `regex` instead of `abc` to make this a bit more obvious that it is testing regex.

Reply:
can you also update the following tests:
1. `TestAuthorization_Conditions` to cover the request.headers and source.principal
2. `TestAuthorization_TCP` to cover the source.principal
3. `TestAuthorization_JWT` to cover jwt claim